### PR TITLE
[WWSTCERT-13550] Sonoff - Add the pressure display function to SNZB-02M

### PR DESCRIPTION
**Pull request by laity-w-sudo**
Check all that apply

# Type of Change

- [x] WWST Certification Request
     - If this is your first time contributing code:
          - [x] I have reviewed the README.md file
          - [x] I have reviewed the CODE_OF_CONDUCT.md file
          - [x] I have signed the CLA
     - [x] I plan on entering a WWST Certification Request or have entered a request through the WWST Certification console at developer.smartthings.com
- [ ] Bug fix
- [x] New feature
- [ ] Refactor

# Checklist

- [x] I have performed a self-review of my code
- [x] I have commented my code in hard-to-understand areas
- [x] I have verified my changes by testing with a device or have communicated a plan for testing
- [x] I am adding new behavior, such as adding a sub-driver, and have added and run new unit tests to cover the new behavior

# Description of Change
Adding the air pressure function to the SNZB-02M equipment

# Summary of Completed Tests

### CODE REVIEW

File: drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/can_handle.lua [hunk 1]
```diff
@@ -0,0 +1,12 @@
+-- Copyright 2025 SmartThings, Inc.
```
Comment (KKlimczukS) [inline drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/can_handle.lua]:
```suggestion
-- Copyright 2026 SmartThings, Inc.
```

File: drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/init.lua [hunk 2]
```diff
@@ -0,0 +1,58 @@
+--[[
+Description: SNZB-02M pressure/humidity/temperature sensor driver
+Version: 1.0
+Author: GitHub Copilot
+--]]
```
Comment (KKlimczukS) [inline drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/init.lua]:
```suggestion
-- Copyright 2026 SmartThings, Inc.
-- Licensed under the Apache License, Version 2.0
```

File: drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/init.lua [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+local log = require "log"
+
+local function convert_pressure_value(raw_value)
+  if raw_value == nil then
+    return nil
+  end
+
+  if raw_value > 2000 then
+    return raw_value / 1000.0
+  end
```
Comment (KKlimczukS) [inline drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/init.lua]: Please, could You add short explanation (in a comment) above this "if" statement regarding the edge case that forces this division ?

File: tools/run_driver_tests_p.py [hunk 4]
```diff
@@ -10,6 +10,8 @@
 DRIVER_DIRS = Path(os.path.abspath(__file__)).parents[1].joinpath("drivers")
 DRIVERS = [driver for driver in DRIVER_DIRS.glob("*/*") if driver.is_dir()] # this gets all the children of the children of the drivers directory
 CHANGED_DRIVERS = [Path(driver).name for driver in sys.argv[1:]]
+if CHANGED_DRIVERS:
+  DRIVERS = [driver for driver in DRIVERS if driver.name in CHANGED_DRIVERS]
```
Comment (KKlimczukS) [inline tools/run_driver_tests_p.py]: This change is not directly related to Your device.

File: tools/run_driver_tests_p.py
Comment (cjswedes) [inline tools/run_driver_tests_p.py:1]: Remove the changes to this file.

**Review by cjswedes — CHANGES_REQUESTED** [review]

File: drivers/SmartThings/zigbee-humidity-sensor/profiles/sonoff-humidity-temp-press-battery.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  - id: atmosphericPressureMeasurement
+    version: 1
+  - id: battery
+    version: 1
+  - id: firmwareUpdate
+    version: 1
+  - id: refresh
+    version: 1
+  categories:
+  - name: MultiFunctionalSensor
```
Comment (KKlimczukS) [inline drivers/SmartThings/zigbee-humidity-sensor/profiles/sonoff-humidity-temp-press-battery.yml:18]:
```suggestion
  - name: HumiditySensor
```

**Review by KKlimczukS — CHANGES_REQUESTED** [review]

File: drivers/SmartThings/zigbee-humidity-sensor/src/test/test_sonoff_snzb02m.lua [hunk 6]
```diff
@@ -0,0 +1,117 @@
+-- Copyright 2025 SmartThings, Inc.
```
Comment (KKlimczukS) [inline drivers/SmartThings/zigbee-humidity-sensor/src/test/test_sonoff_snzb02m.lua:1]:
```suggestion
-- Copyright 2026 SmartThings, Inc.
```

File: drivers/SmartThings/zigbee-humidity-sensor/src/sub_drivers.lua [hunk 7]
```diff
@@ -9,5 +9,6 @@ local sub_drivers = {
    lazy_load_if_possible("centralite-sensor"),
    lazy_load_if_possible("heiman-sensor"),
    lazy_load_if_possible("frient-sensor"),
+   lazy_load_if_possible("sonoff.SNZB-02M"),
```
Comment (KKlimczukS) [inline drivers/SmartThings/zigbee-humidity-sensor/src/sub_drivers.lua:12]: Please remove the comma at  the end of line

File: drivers/SmartThings/zigbee-humidity-sensor/profiles/sonoff-humidity-temp-press-battery.yml (hunk 5 above)
Comment (KKlimczukS) [reply drivers/SmartThings/zigbee-humidity-sensor/profiles/sonoff-humidity-temp-press-battery.yml:18]: @laity-w-sudo this requested change is mandatory

File: drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/init.lua (hunk 3 above)
Comment (KKlimczukS) [reply drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/init.lua]: @laity-w-sudo How come this change was removed ? Did You change anything on device's firmware ?

File: drivers/SmartThings/zigbee-humidity-sensor/src/sub_drivers.lua (hunk 7 above)
Comment (laity-w-sudo) [reply drivers/SmartThings/zigbee-humidity-sensor/src/sub_drivers.lua]: > Please remove the comma at the end of line

Hi, the modifications have been made according to your suggestions. However, due to the changes, the automated production line test has failed at present. How can we solve this problem? Also, the error message seems to have nothing to do with the driver I submitted.

File: drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/init.lua (hunk 3 above)
Comment (KKlimczukS) [reply drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/init.lua]: @laity-w-sudo ?

File: drivers/SmartThings/zigbee-humidity-sensor/profiles/sonoff-humidity-temp-press-battery.yml (hunk 5 above)
Comment (KKlimczukS) [reply drivers/SmartThings/zigbee-humidity-sensor/profiles/sonoff-humidity-temp-press-battery.yml]: @laity-w-sudo please change the category here to HumiditySensor

File: drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/init.lua (hunk 3 above)
Comment (laity-w-sudo) [reply drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/init.lua]: > @laity-w-sudo How come this change was removed ? Did You change anything on device's firmware ?The device firmware has not been modified. This part of the logic is not very useful, so I have removed it.

File: drivers/SmartThings/zigbee-humidity-sensor/src/test/test_sonoff_snzb02m.lua (hunk 6 above)
Comment (KKlimczukS) [reply drivers/SmartThings/zigbee-humidity-sensor/src/test/test_sonoff_snzb02m.lua]: @laity-w-sudo please change this one according to the suggestion

Comment (laity-w-sudo) [reply drivers/SmartThings/zigbee-humidity-sensor/src/test/test_sonoff_snzb02m.lua]: > @laity-w-sudo please change this one according to the suggestion

The latest submission has already been revised according to your suggestions.

Comment (KKlimczukS) [reply drivers/SmartThings/zigbee-humidity-sensor/src/test/test_sonoff_snzb02m.lua]: @laity-w-sudo looks like that line remained unchanged.

Comment (laity-w-sudo) [reply drivers/SmartThings/zigbee-humidity-sensor/src/test/test_sonoff_snzb02m.lua]: > @laity-w-sudo looks like that line remained unchanged.

I made the modifications in commit b7123f5a52ba65167e09ce88bddf0d14fff3d2b0, as shown in the figure.

Comment (KKlimczukS) [reply drivers/SmartThings/zigbee-humidity-sensor/src/test/test_sonoff_snzb02m.lua]: @laity-w-sudo The screenshot comes from can_handle.lua.

The request was also related to this file -> test_sonoff_snzb02m.lua‎

File: drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/init.lua [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+local capabilities = require "st.capabilities"
+local clusters = require "st.zigbee.zcl.clusters"
+local log = require "log"
+
+local function convert_pressure_value(raw_value)
+  if raw_value == nil then
+    return nil
+  end
+
+  return raw_value / 10.0
```
Comment (cjswedes) [inline drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/init.lua:24]: This currently fails CI, `convert_pressure_value` always divides by 10, but test_sonoff_snzb02m.lua expects raw values above 2000 to be divided by 1000. You removed the threshold branch in a later commit but the tests still assume it exists. What is the correct functionality? Please either restore the conditional (with a comment explaining the firmware/scale edge case) or fix the tests so they match the actual conversion you intend to ship

File: drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/can_handle.lua [hunk 9]
```diff
@@ -0,0 +1,12 @@
+-- Copyright 2026 SmartThings, Inc.
+-- Licensed under the Apache License, Version 2.0
+
+local function can_handle(opts, driver, device, ...)
+  if device:get_model() == "SNZB-02M" then
```
Comment (cjswedes) [inline drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/can_handle.lua]: This check should also be checking for `device:get_manaufacturer() == "SONOFF"`

File: drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/can_handle.lua
Comment (cjswedes) [inline drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/can_handle.lua:1]: Is there a reason to have the nested directories? We should just add a `sonoff` subdriver rather than having `sonoff/SNZB-02M`. We want to avoid having single subdrivers for each model.

File: drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/init.lua [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+end
+
+local can_handle = require "sonoff.SNZB-02M.can_handle"
+
+return {
+  NAME = "Sonoff SNZB-02M Sensor",
+  zigbee_handlers = {
+    attr = {
+      [clusters.PressureMeasurement.ID] = {
+        [clusters.PressureMeasurement.attributes.MeasuredValue.ID] = pressure_report_handler
```
Comment (cjswedes) [inline drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/init.lua]: Did you test that the PressureMeasurement attribute is reported automatically when the value changes? I do not see any reporting configuration for the attribute or binding of the cluster. I would expect to see a call in a device_init handler to `device:add_configured_attribute` to ensure this configuration is setup when the device is configured.

**Review by cjswedes — COMMENTED** [review]
My biggest concern is the lack of configuration for reporting the PressureMeasurement MeasuredValue attribute; has this been tested manually?

File: drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/can_handle.lua
Comment (laity-w-sudo) [reply drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/can_handle.lua]: > Is there a reason to have the nested directories? We should just add a `sonoff` subdriver rather than having `sonoff/SNZB-02M`. We want to avoid having single subdrivers for each model.
We have a large number of sub-devices. I wonder if only one driver is used for management, won't these files become very cumbersome? We create folders to facilitate the management of the devices.

File: drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/init.lua [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+end
+
+local can_handle = require "sonoff.SNZB-02M.can_handle"
+
+return {
+  NAME = "Sonoff SNZB-02M Sensor",
+  zigbee_handlers = {
+    attr = {
+      [clusters.PressureMeasurement.ID] = {
+        [clusters.PressureMeasurement.attributes.MeasuredValue.ID] = pressure_report_handler
```
Comment (laity-w-sudo) [reply drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/init.lua]: May I ask if this is an essential action? My device has already been equipped with the default pressure reporting function at the factory. The default reporting configuration is exactly what we want, and there is no need to issue a new configuration.

File: drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/init.lua (hunk 10 above)
Comment (cjswedes) [reply drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/init.lua]: Most clusters get configured due to default functionality, but I dont believe we are configuring this one with the defaults, so the subdriver should be doing so.

File: drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/can_handle.lua
Comment (cjswedes) [reply drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/SNZB-02M/can_handle.lua:1]: Please change the directory structure so that it is not nested, and everything falls under the `sonoff` directory.

File: drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/init.lua (hunk 10 above)
Comment (laity-w-sudo) [reply drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/init.lua]: > Most clusters get configured due to default functionality, but I dont believe we are configuring this one with the defaults, so the subdriver should be doing so.

Yes, the default behavior of the equipment already includes the configuration for pressure reporting. Is it still necessary to explicitly add it in the driver?

Comment (KKlimczukS) [reply drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/init.lua]: @laity-w-sudo 
here is the example of cluster configuration in the subdriver:
https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/71bbd3da8178b1c38332a6d9342ea23a3ce5b3dd/drivers/SmartThings/zigbee-power-meter/src/frient/EMIZB-151/init.lua#L98

Comment (KKlimczukS) [reply drivers/SmartThings/zigbee-humidity-sensor/src/sonoff/init.lua]: @laity-w-sudo Please add configuration for this cluser/attribute according to suggestions

**Review by KKlimczukS — CHANGES_REQUESTED** [review]
@laity-w-sudo  Please update Your branch.

**Review by KKlimczukS — APPROVED** [review]
lgtm

File: drivers/SmartThings/zigbee-humidity-sensor/src/test/test_sonoff_snzb02m.lua [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+
+zigbee_test_utils.prepare_zigbee_env_info()
+local function test_init()
+  test.mock_device.add_test_device(mock_device)
+end
+
+test.set_test_init_function(test_init)
+
+test.register_message_test(
+  "Pressure report at or below 2000 should be divided by 10",
```
Comment (cjswedes) [inline drivers/SmartThings/zigbee-humidity-sensor/src/test/test_sonoff_snzb02m.lua:37]: What is the functionality for reports above 2000? Is it supposed to be different?

If there is no difference (which is what is implemented now), then please update the name of this test:

```suggestion
  "Pressure report should be divided by 10",
```

If there is different handling of the report if the value is >2000, please implement and write a new test.
